### PR TITLE
Fix #2559: Fix throwable message for this(cause: Throwable) constructor.

### LINKED ISSUE #2559: Throwable constructor with Throwable not in spec.
just to report...
https://github.com/scala-js/scala-js/blob/master/javalanglib/src/main/scala/java/lang/Throwables.scala#L330

in javadocs I read 
```Constructs a new exception with the specified cause and a detail message of (cause==null ? null : cause.toString()) (which typically contains the class and detail message of cause).```
https://docs.oracle.com/javase/7/docs/api/java/lang/ReflectiveOperationException.html#ReflectiveOperationException(java.lang.Throwable)

that is not the case.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd

**Comment by andreaTP** [comment]
:+1:  lazy august :-)

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala [hunk 1]
```diff
@@ -63,4 +63,55 @@ class ThrowablesTest {
...
+      assertNull(newThrowable3(null, t1).getMessage)
+
+      assertEquals(newThrowable1.getClass.getName,
+          newThrowable4(t0).getMessage)
+      assertEquals(newThrowable1.getClass.getName + ": foo",
+          newThrowable4(t1).getMessage)
+    }
+
+    test(new Throwable, s => new Throwable(s), (s, t) => new Throwable(s, t),
+        t => new Throwable(t))
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala]: Consider using this style:
```scala
test(new Throwable(), new Throwable(_), new Throwable(_), new Throwable(_, _))
```

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala [hunk 2]
```diff
@@ -63,4 +63,55 @@ class ThrowablesTest {
...
+      val t1 = newThrowable2("foo")
+      assertNull(t0.getMessage)
+      assertEquals("foo", t1.getMessage)
+
+      assertEquals("bar", newThrowable3("bar", t0).getMessage)
+      assertEquals("bar", newThrowable3("bar", t1).getMessage)
+      assertNull(newThrowable3(null, t0).getMessage)
+      assertNull(newThrowable3(null, t1).getMessage)
+
+      assertEquals(newThrowable1.getClass.getName,
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala]: Is calling `newThrowable1` every time intentional? Reuse `t0` instead?

**Comment by sjrd** [comment]
This is missing the throwables in other packages (`java.io`, `java.net`, `java.util`, `java.security`, `java.util.concurrent`).

**Comment by sjrd** [comment]
That's all.

**Comment by nicolasstucki** [comment]
I only added the throwables in if they have the `def this(e: Throwable)` constructor.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala [hunk 3]
```diff
@@ -63,4 +63,93 @@ class ThrowablesTest {
...
+    val t1 = new Throwable("foo")
+
+    def test0(newThrowable: Throwable): Unit = {
+      assertNull(newThrowable.getMessage)
+    }
+
+    def test1(newThrowable: String => Throwable): Unit = {
+      assertEquals("foo", newThrowable("foo").getMessage)
+    }
+    def test2(newThrowable: Throwable => Throwable): Unit = {
```
Comment (sjrd) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala]: Missing blank line above this line.

**Comment by sjrd** [comment]
That's all.

File: test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala (hunk 3 above)
Comment (nicolasstucki) [inline test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ThrowablesTest.scala]: Fixed

**Comment by sjrd** [comment]
LGTM

**Comment by sjrd** [comment]
Jenkins, retest this please

**Comment by sjrd** [comment]
LGTM

**Pull request merged**